android/tangram/src/com/mapzen/tangram/Marker.java

```diff
@@ -0,0 +1,179 @@
+package com.mapzen.tangram;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.util.DisplayMetrics;
+
+import com.mapzen.tangram.geometry.Polygon;
+import com.mapzen.tangram.geometry.Polyline;
+
+/**
+ * Class used to display points, polylines, and bitmaps dynamically on a map. Do not create one of
+ * these objects directly, instead use {@link MapController#addMarker()}.
+ */
+public class Marker {
+
+    Context context;
+    long markerId = 0;
+    MapController map;
+    boolean visible = true;
+
+    /**
+     * Package private constructor for creating a new {@link Marker}.
+     * @param context the context to use for decoding bitmap resources
+     * @param markerId the marker id
+     * @param map the map this marker is added to
+     */
+    Marker(Context context, long markerId, MapController map) {
+        this.context = context;
+        this.markerId = markerId;
+        this.map = map;
+    }
+
+    /**
+     * Sets the styling to be used to display either a point, polyline, or bitmap for this marker.
+     * If the marker is going to be used to display a bitmap, a 'points' style must be set.
+     *
+     * ie. { style: 'points', color: 'white', size: [50px, 50px], order: 2000, collide: false }
```

Comment:
It would be nice to add some formatting tags here otherwise these examples will display as one long paragraph when the javadoc is parsed.

```
   * <ul>
   * <li>{ style: 'points', color: 'white', size: [50px, 50px], order: 2000, collide: false }</li>
   * <li>{ style: 'lines', color: '#06a6d4', width: 5px, order: 2000 }</li>
   * <li>{ style: 'polygons', color: '#06a6d4', width: 5px, order: 2000 }</li>
   * </ul>
```